nixos/modules/services/network-filesystems/openafs/server.nix

```diff
@@ -296,13 +391,16 @@ in {
         after = [ "network.target" ];
         wantedBy = [ "multi-user.target" ];
         restartIfChanged = false;
-        unitConfig.ConditionPathExists = [
+        unitConfig.ConditionPathExists = mkIf (cfg.authentication.keytab == null) [
           "|/etc/openafs/server/KeyFileExt"
         ];
         preStart = ''
           mkdir -m 0755 -p /var/openafs
           ${optionalString (netInfo != null) "cp ${netInfo} /var/openafs/netInfo"}
           ${optionalString useBuCellServDB "cp ${buCellServDB}"}
+        '' + lib.optionalString (cfg.authentication.keytab != null) ''
+          rm -f /etc/openafs/server/KeyFileExt
+          ${asetkeyScript}
```

Comment:
Create `/etc/openafs/server/KeyFileExt` with proper access rights, first (e.g. 600). I am not sure whether asetkey does that on its own or just uses the current umask.